web_monitoring/diffing_server.py

```diff
@@ -115,8 +115,35 @@ def options(self):
 class DiffHandler(BaseHandler):
     # subclass must define `differs` attribute
 
+    # Compute our own ETag header values.
+    # We're not actually hashing content for this, since that is expensive.
+    def compute_etag(self):
+        query_params = {k: v[-1].decode() for k, v in
+            self.request.arguments.items()}
+        etag = str('W/"' + hashlib.sha256(
+            web_monitoring.__version__.encode('utf-8') + self.request.path.encode('utf-8') + str(query_params).encode('utf-8')
+        ).hexdigest() + '"').encode('utf-8')
```

Comment:
This seems slightly unwieldy to read — maybe split up the hash composition and the string?

```py
hash = hashlib.sha256(
    web_monitoring.__version__.encode('utf-8') +
    self.request.path.encode('utf-8') +
    str(query_params).encode('utf-8'))
return f'W/"{hash.hexdigest()}"'.encode('utf-8')
```

(Side note: does this methods actually require us to return bytes instead of a string?)